src/java.desktop/unix/native/libawt_wlawt/WLComponentPeer.c

```diff
@@ -467,6 +488,16 @@ Java_sun_awt_wl_WLComponentPeer_nativeCreateWLSurface
     if (parentFrame && parentFrame->toplevel) {
         xdg_toplevel_set_parent(frame->xdg_toplevel, parentFrame->xdg_toplevel);
     }
+    if (zxdg_decoration_manager_v1 != NULL) {
```

Comment:
Is there any benefit in doing this when `isUndecorated` is true? Seems like we can simply skip all the code under `if` here.

Reply:
Not declaring a preference would cause compositors that prefer SSDs by default to use SSDs on such windows.

Reply:
> Not declaring a preference would cause compositors that prefer SSDs by default to use SSDs on such windows.

I haven't noticed that on KDE and the specification does not say that. What are those compositors that enforce SSD?

Reply:
>I haven't noticed that on KDE and the specification does not say that.

Which specification? If you're not using this protocol for a window, then nothing in this protocol applies to the window and the compositor decorates as it sees fit without the ability to tell the client what that decoration is.

Reply:
>Is there a downside to that?

If the compositor decides to use SSD even on windows that the JDK wants to be undecorated, e.g. the shift-shift window in IntelliJ, then, if you're not using this protocol, the compositor has no way to tell the JDK that SSDs are being drawn.

That would not change anything with the current state of this PR but it makes some improvements impossible. For example, the shift-shift window currently enters resize-mode when the cursor is near the edges of the window. However, resizing will not work if the compositor uses SSDs even for the shift-shift window and the user should use the SSDs to resize the window instead.

Reply:
Done